.github/workflows/lint.yml

```diff
@@ -0,0 +1,62 @@
+## SECURITY WARNING:
+##
+## Do not change this job unless you know what you're doing.
+##
+## This GitHub Action runs on: pull_request_target, which means the jobs run in
+## a context where they have access to a Access Token with Read access to the
+## target repo, even if the PR is opened from an external contributor from their
+## fork.
+##
+## This means that if we're not careful, we could be running third-party code
+## within an authenticated scope, which isn't good. To mitigate this, this
+## implementation does:
+##
+##   1. checkout the target branch (i.e. the project's original sources)
+##   2. install the Gems from there, and install them into a directory that's
+##      outside the repository contents.
+##   3. checkout the PRs HEAD
+##   4. restore a bunch of files that would allow code execution from the
+##      project's upstream sources, namely:
+##      - bin/bundle - we'll run that in our Job
+##      - Gemfile/Gemfile.lock - to avoid loading a gem with an identical
+##        version number from a in-repo vendored directory
+
+name: Lint
+on:
+  pull_request_target:
+
+permissions:
+  contents: read
+  statuses: write
+  pull-requests: write
+
+jobs:
+  pronto:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install system dependencies
+        run: sudo apt update && sudo apt install -y build-essential curl git gsfonts imagemagick libcurl4-openssl-dev libidn11-dev libmagickwand-dev libssl-dev libxml2-dev libxslt1-dev yarnpkg
+      - name: Checkout Target branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+          fetch-depth: 0
```

Comment:
The `fetch-depth: 0` isn't required, it's enough if the latest commit (so the default) of the target branch is there, and the commit of the PR-branch is fetched bellow. These are the only commits pronto needs to work.

Reply:
Note what might be confusing here: the checkout action is *not* doing a shallow clone. It's `git init`'ing a completely empty repo, then adding the origin repo as a remote, then it's `git fetch`'ing the rev you specified. You can actually reproduce this failure locally:

In a completely new directory:

```
# git init
Initialized empty Git repository in /Users/denschub/Projects/diaspora/tmp/.git/

# git remote add origin file:///Users/denschub/Projects/diaspora/diaspora/
git fetch --no-tags --prune --depth=1 origin '+refs/heads/develop*:refs/remotes/origin/develop*'
remote: Enumerating objects: 2183, done.
remote: Counting objects: 100% (2183/2183), done.
remote: Compressing objects: 100% (2076/2076), done.
remote: Total 2183 (delta 141), reused 959 (delta 53), pack-reused 0
Receiving objects: 100% (2183/2183), 10.33 MiB | 40.24 MiB/s, done.
Resolving deltas: 100% (141/141), done.
From file:///Users/denschub/Projects/diaspora/diaspora
 * [new branch]      develop    -> origin/develop

# git checkout --progress --force -B develop refs/remotes/origin/develop
branch 'develop' set up to track 'origin/develop'.
Switched to a new branch 'develop'

#  git log -1 --format='%H'
6430aa3bad6fb4f85765b380aaf92fc2deff0c10
```

Then, create a new branch and make a change:

```
# git checkout -b tmp
Switched to a new branch 'tmp'

# rm README.md

# git commit -a -m "foo"
[tmp 8946483] foo
 1 file changed, 31 deletions(-)
 delete mode 100644 README.md
```

and now you're essentially where the workflow would be after fetching the PR head.

If you now try to run Pronto locally, it'll fail the exact same way:

```
# bin/pronto run -c develop
warning: parser/current is loading parser/ruby31, which recognizes3.1.2-compliant syntax, but you are running 3.1.4.
Please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
/Users/denschub/.gem/ruby/3.1.4/gems/pronto-0.11.0/lib/pronto/git/repository.rb:87:in `merge_base': object not found - no match for id (3b07b1696211cc489f257f9136ce67c265e1ee33) (Rugged::OdbError)
```

Reply:
Interesting, I actually tested this on my fork (because I wasn't sure, but couldn't think of a reason why it should work), and I didn't have any problems. Does it only break in some cases? I still don't fully understand why it shouldn't work? :thinking: 

> Unfortunately, running pronto after that fails

Did you also fetch and checkout the PR-branch you want to compare against `main` before running pronto? Or are you trying to compare `main` against `main`?

Reply:
OK, after some more testing, I also managed to fail it ... so I have still no idea why my first test worked, after my understanding what pronto does based on your explanation, my first test should also have failed :man_shrugging: